crates/provider-http-server/tests/links.rs

```diff
@@ -0,0 +1,194 @@
+    )
+    .await?;
+    assert_advertise_link(
+        &ctl_client,
+        provider_http_id,
+        component_jsonify_rust_id,
+        "jsonify",
+        "wasi",
+        "http",
+        vec!["incoming-handler".into()],
+        vec![provider_http_config_name.into()],
+        Vec::with_capacity(0),
+    )
+    .await?;
+
+    // Start the provider and components
+    let (_, _, _) = try_join!(
+        assert_start_provider(StartProviderArgs {
+            client: &ctl_client,
+            host_id: &host_id,
+            provider_id: provider_http_id,
+            provider_ref: PROVIDER_HTTP_SERVER_IMAGE_REF,
+            config: vec![provider_http_config_name.into()],
+        }),
+        assert_scale_component(
+            &ctl_client,
+            host_id.as_ref(),
+            COMPONENT_HELLO_WORLD_RUST_IMAGE_REF,
+            component_hello_rust_id,
+            None,
+            1,
+            Vec::new(),
+            Duration::from_secs(10),
+        ),
+        assert_scale_component(
+            &ctl_client,
+            host_id.as_ref(),
+            COMPONENT_HTTP_JSONIFY_RUST_IMAGE_REF,
+            component_jsonify_rust_id,
+            None,
+            1,
+            Vec::new(),
+            Duration::from_secs(10),
+        ),
+    )
+    .context("failed to start provider and components")?;
+
+    // Wait until we can get any response from the HTTP server
+    wasmcloud_test_util::http::wait_for_url(&endpoint_url)
+        .await
+        .context("Provider HTTP URL was never available")?;
+
+    // Perform a request which should go to the rust component ('default' link)
+    let get_lowercased_resp = || async {
+        reqwest::get(&endpoint_url)
+            .await
+            .context("failed to get /")?
+            .text()
+            .await
+            .map(|s| s.to_lowercase())
+            .context("failed to get textual output of request")
+    };
+    assert!(
+        get_lowercased_resp().await?.contains("hello from rust"),
+        "first response should be from rust component"
+    );
+
+    // Delete the default link
+    assert_remove_link(&ctl_client, provider_http_id, "wasi", "http", "default")
+        .await
+        .context("failed to remove link 'default'")?;
+
+    // At this point, the provider should *STILL* be pointing to the rust component,
+    // as it was set *second* ('hello' link)
+    assert!(
+        get_lowercased_resp().await?.contains("hello from rust"),
+        "second response should be from rust component"
+    );
+
+    // Delete the rust link
+    assert_remove_link(&ctl_client, provider_http_id, "wasi", "http", "rust")
+        .await
+        .context("failed to remove link 'rust'")?;
+
+    // At this point, the provider should be pointing to the jsonify component,
+    // as it was set *third* ('jsonify' link)
+    assert!(
+        !get_lowercased_resp().await?.contains("hello from rust"),
+        "third response be from jsonify component",
+    );
```

Comment:
This feels like a big behavior change, I can see the appeal to enable the other link getting put initially while you clean up older applications. I think once we have a mechanism to report failed links back to the host, we probably shouldn't enable this scenario to happen (in address based mode). I'm on the fence on if this is nice because the provider can't retry putting a link after it already failed (today) or if this will lead to confusing scenarios.

Any thoughts here? I could go either way here I think. Curious what spurred this change initially and I think that would help me understand